app/src/message/message-reactions.component.js

```diff
@@ -0,0 +1,77 @@
+import { Fragment, h } from "preact";
+import { useState, useEffect } from "preact/hooks";
+import { http } from "/src/core";
+import MessageEmojiSelector from "./message-emojiselector.component";
+
+export default function MessageReactions(props) {
+  const [reactions, setReactions] = useState([]);
+  const [hoveredReaction, setHoveredReaction] = useState(null);
+
+  const MAX_VISIBLE_USERS = 10;
+
+  const loadReactions = async (reactionData) => {
+    const formattedReactions = Object.values(reactionData).map(reaction => ({
+      emoji: reaction.emoji,
+      count: reaction.count,
+      users: reaction.users,
+      currentUserReactionId: reaction.currentUserReactionId,
+    }));
+    setReactions(formattedReactions);
+  };
+
+  useEffect(() => {
+    if (!props.messageId) return;
+
+    const fetchReactions = async () => {
+      const reactionData = await http.get(`/api/messages/${props.messageId}/reactions`);
+      await loadReactions(reactionData);
+    };
+
+    fetchReactions();
+  }, [props.messageId]);
+
+  const handleReactionClick = async (currentUserReactionId) => {
+    // If current user set this emoji type, click to remove it
+    if (currentUserReactionId) {
+      await http.delete(`/api/messages/${props.messageId}/reactions/${currentUserReactionId}`);
+      await loadReactions(await http.get(`/api/messages/${props.messageId}/reactions`));
+    }
+  };
+
+  return (
+    <a className="d-flex action seamless-link font-size-90 capitalize" >
+      {reactions.map(({ emoji, count, users, currentUserReactionId: currentUserReactionId }) => (
+        <div
+          key={emoji}
+          className="reaction-emoji"
+          style={{
+            cursor: currentUserReactionId ? "pointer" : "default",
+          }}
+          title={users.slice(0, MAX_VISIBLE_USERS).join("\n") + (users.length > MAX_VISIBLE_USERS ? `\nand ${users.length - MAX_VISIBLE_USERS} more...` : "")}
```

Comment:
I saw that while editing on my branch but thought to leave for later (so now): we would need to add a translation string for this. It could have only a key for english for now though.

Reply:
Good catch! I've just added the en_US translation and the call to the translation function in a new commit.